account_billing/models/account_billing.py

```diff
@@ -0,0 +1,194 @@
+        default=lambda self: self._get_partner_id(),
+        help='Partner Information',
+        track_visibility='always'
+    )
+    date = fields.Date(
+        string='Billing Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: time.strftime('%Y-%m-%d'),
+        help='Effective date for accounting entries',
+        track_visibility='always'
+    )
+    due_date = fields.Date(
+        string='Due Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: fields.Date.context_today(self),
+        required=True,
+        track_visibility='always'
+    )
+    invoice_ids = fields.One2many(
+        'account.invoice',
+        'billing_id',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self._context.get('active_ids', []),
+        string='Invoices',
+    )
+    invoice_related_count = fields.Integer(
+        string='# of Invoice',
+        compute='_compute_invoice_related_count',
+        help='Count invoice in billing'
+    )
+    state = fields.Selection(
+        [('draft', 'Draft'),
+         ('cancel', 'Cancelled'),
+         ('billed', 'Billed')],
+        string='Status',
+        readonly=True,
+        default='draft',
+        help="""* The 'Draft' status is used when a user is encoding a new and
+            unconfirmed billing.
+            \n* The 'Billed' status is used when user create billing,
+            a billing number is generated
+            \n* The 'Cancelled' status is used when user cancel billing.""",
+    )
+    narration = fields.Text(
+        string='Notes',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="Notes"
+    )
+    type = fields.Selection(
+        [('out_invoice', 'Customer Invoice'),
+         ('in_invoice', 'Vendor Bill')],
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        index=True,
+        change_default=True,
+        default=lambda self: self._context.get('type', False)
+    )
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form',
+                        toolbar=False, submenu=False):
+        res = super(AccountBilling, self).fields_view_get(
+            view_id=view_id, view_type=view_type,
+            toolbar=toolbar, submenu=submenu)
+        active_ids = self._context.get('active_ids', False)
+        invoice_ids = self.env['account.invoice'].search([
+            ('id', 'in', active_ids)])
+        billing_ids = invoice_ids.mapped('billing_id')
+        if any(inv.state != 'open' for inv in invoice_ids):
+            raise ValidationError(_("The billing cannot be processed \
+                because the invoice is not open!"))
+        if billing_ids:
+            raise ValidationError(_("Invoice(s) create billing already!"))
+        return res
+
+    @api.onchange('type')
+    def _onchange_partner_id(self):
```

Comment:
Can we change onchange name?
```suggestion
    def _onchange_type(self):
```